linux/devdoc/async_socket_linux_requirements.md

```diff
@@ -275,62 +233,58 @@ MOCKABLE_FUNCTION(, int, async_socket_receive_async, ASYNC_SOCKET_HANDLE, async_
 
 **SRS_ASYNC_SOCKET_LINUX_11_074: [** The context shall also allocate enough memory to keep an array of `buffer_count` items. **]**
 
-**SRS_ASYNC_SOCKET_LINUX_11_075: [** `async_socket_receive_async` shall add the socket in the epoll system by calling `epoll_ctl` with `EPOLL_CTL_MOD` **]**
-
-**SRS_ASYNC_SOCKET_LINUX_11_076: [** If the `epoll_ctl` call fails with `ENOENT`, `async_socket_send_async` shall call `epoll_ctl` again with `EPOLL_CTL_ADD`. **]**
-
 **SRS_ASYNC_SOCKET_LINUX_11_077: [** On success, `async_socket_receive_async` shall return 0. **]**
 
 **SRS_ASYNC_SOCKET_LINUX_11_078: [** If any error occurs, `async_socket_receive_async` shall fail and return a non-zero value. **]**
 
-### thread_worker_func
+### event_complete_callback
 
 ```c
-static int thread_worker_func(void* parameter)
+static void event_complete_callback(void* context, COMPLETION_PORT_EPOLL_ACTION action)
 ```
 
-`thread_worker_func` handles the epoll thread.
+`event_complete_callback` handles all the completion port callback info.
 
-**SRS_ASYNC_SOCKET_LINUX_11_079: [** `thread_worker_func` shall call `epoll_wait` waiting for the epoll to become signaled. **]**
+**SRS_ASYNC_SOCKET_LINUX_11_079: [** If context is `NULL`, `event_complete_callback` shall do nothing. **]**
 
-**SRS_ASYNC_SOCKET_LINUX_11_080: [** Onced signaled `thread_worker_func` shall loop through the signaled epolls. **]**
+**SRS_ASYNC_SOCKET_LINUX_11_080: [** If `COMPLETION_PORT_EPOLL_ACTION` is `COMPLETION_PORT_EPOLL_EPOLLRDHUP` or `COMPLETION_PORT_EPOLL_ABANDONED`, `event_complete_callback` shall do the following: **]**
 
-**SRS_ASYNC_SOCKET_LINUX_11_081: [** If the events value contains `EPOLLRDHUP` (hang up), `thread_worker_func` shall the following: **]**
+- **SRS_ASYNC_SOCKET_LINUX_11_081: [** `event_complete_callback` shall call either the send or recv complete callback with an `ABANDONED` flag. **]**
 
-- **SRS_ASYNC_SOCKET_LINUX_11_082: [** `thread_worker_func` shall receive the `ASYNC_SOCKET_RECV_CONTEXT` value from the ptr variable from the `epoll_event` data ptr. **]**
+- **SRS_ASYNC_SOCKET_LINUX_11_084: [** Then `event_complete_callback` shall and free the `io_context` memory. **]**
```

Comment:
>io_context [](http://example.com/codeflow?start=90&length=10)

The only reference to this `io_context` is right in this function (`event_complete_callback`). Would be nice to know what's the `io_context` supposed to contain/how it is used /etc